lib/Controller/AccountsController.php

```diff
@@ -186,6 +186,7 @@ public function update($id, $accountName, $emailAddress, $password, $imapHost, $
 	 * @return JSONResponse
 	 */
 	public function destroy($id) {
+		$this->aliasesService->deleteAll($id, $this->currentUserId);
 		$this->accountService->delete($this->currentUserId, $id);
```

Comment:
I'd prefer to hide this logic here and rather do it in the account service.

Reply:
I agree, I wanted to put it into `AccountService` as well, but could not find a way to call `AliasesService` from there. Did I miss something?

Reply:
I'd suggest to inject the alias mapper in the account service.

Reply:
so, extending the constructor with an additional argument as well?

Reply:
Exactly

Reply:
OK, I just wanted to make sure I don't blow up the constructor further needlessly.